### PR TITLE
Component refactor

### PR DESCRIPTION
**Pull request by NateCastleOIT**
Refactored the Player's input, physics, and graphics into their own components. Those are derived from generic respective component classes. This will be the main theme moving forward. Want a new enemy? Make new components that will be passed during object creation that will determine its behavior.

### CODE REVIEW

File: TKGame/Components/Concrete/Player_PhysicsComponent.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using TKGame.Components.Interface;
+
+namespace TKGame.Components.Concrete
+{
+    internal class Player_PhysicsComponent : PhysicsComponent
+    {
+        private static readonly float GRAVITY = 1.0f;
+        void PhysicsComponent.Update(Entity /*&*/entity, GameTime gameTime/*, World &world*/) // The &reference isn't working.
```
Comment (keeganhauser) [inline TKGame/Components/Concrete/Player_PhysicsComponent.cs:14]: hope this helps 👍 https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/ref

**Review by keeganhauser — APPROVED** [review]
looks chill homeslice

File: TKGame/Components/Concrete/Player_PhysicsComponent.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TKGame.Components.Interface;
+
+namespace TKGame.Components.Concrete
+{
+    internal class Player_PhysicsComponent : PhysicsComponent
+    {
+        private static readonly float GRAVITY = 1.0f;
```
Comment (NathanGr33n) [inline TKGame/Components/Concrete/Player_PhysicsComponent.cs:13]: We may want to adjust the gravity value in the future to actually keep the entities on the platform most of the time. Right now they can just run up vertically, overcoming gravity. However, this is fine for now.

**Review by NathanGr33n — APPROVED** [review]
Looks good for now, some future changes that may need to be made. Code could benefit from some more comments.

File: TKGame/Components/Concrete/Player_PhysicsComponent.cs (hunk 1 above)
Comment (NateCastleOIT) [reply TKGame/Components/Concrete/Player_PhysicsComponent.cs:14]: Yeah, I'm a silly guy who forgot to just look it up

File: TKGame/Components/Concrete/Player_PhysicsComponent.cs (hunk 2 above)
Comment (NateCastleOIT) [reply TKGame/Components/Concrete/Player_PhysicsComponent.cs:13]: Potentially, yes, but that's only if we give them the ability to simply move vertically without jumping, which I don't think is the plan

**Pull request merged**